src/engine/Settings.cs

```diff
@@ -395,7 +410,11 @@ private void CopySettings(Settings settings)
             if (!property.CanWrite)
                 continue;
 
-            property.SetValue(this, property.GetValue(settings));
+            // Since the properties we want to copy are SettingValue generics we use the IAssignableSetting
+            // interface and AssignFrom method to convert the property to the correct concrete class.
+            var setting = property.GetValue(this) as IAssignableSetting;
```

Comment:
I think it would be more proof against programming mistakes in the future if this was:
`(IAssignableSetting)property.GetValue(this)`

so that if someone adds an incorrect property to this class they would get an exception instead of just silently not working correctly.

Reply:
That seems a bit weird to me since there might come a point where you might want to add a property that isn't a SettingValue? But I guess I can change it to that if you really want.

Reply:
I was going to suggest that you move to using the PropertiesOf method I wrote for getting just the JSON saved properties. Instead I just said to add a little bit of future protection in the code for a case where someone accidentally adds a wrong type property.